pdir/api.py

```diff
@@ -198,17 +199,14 @@ def get_oneline_doc(self) -> str:
                 if hasattr(attr, '__delete__'):
                     doc_list.append(DELETER)
                 doc_list[0] = ' '.join([doc_list[0], 'with', doc_list.pop(1)])
-            doc = inspect.getdoc(attr)
-            if doc is not None:
-                doc_list.append(doc.split('\n', 1)[0])
+            doc = get_first_sentence_of_docstring(attr)
+            if doc:
+                doc_list.append(doc)
             return ', '.join(doc_list)
 
         try:
             hasattr_doc = hasattr(attr, '__doc__')
         except:
             hasattr_doc = False
 
-        if hasattr_doc:
-            doc = inspect.getdoc(attr)
-            return doc.split('\n', 1)[0] if doc else ''  # default doc is None
-        return ''
+        return get_first_sentence_of_docstring(attr) if hasattr_doc else ''
```

Comment:
can we call `get_first_sentence_of_docstring(attr)` only once, maybe at the start.

also, I'm wondering whether

```
        try:
            hasattr_doc = hasattr(attr, '__doc__')
        except:
            hasattr_doc = False
```

is still needed.

Reply:
@laike9m I believe the `try` code is from [here](https://github.com/laike9m/pdir2/pull/57) thus still necessary, while it could be integrated with `get_first_sentence_of_docstring`, sounds good?

Reply:
Yes, that sounds good to me. thanks.